File: examples_android/app/src/main/java/com/example/myapp/fragments/viewmodels/models/SurveyModel.kt
package com.example.myapp.fragments.viewmodels.models

import com.confirmit.mobilesdk.SurveySDK
import com.confirmit.mobilesdk.database.externals.Survey
import com.confirmit.mobilesdk.database.externals.SurveyLanguage
import java.io.Serializable

class SurveyModel : Serializable {
    var survey: Survey
        private set
    var selectedLanguage: SurveyLanguage

    constructor(serverId: String, surveyId: String) {
        val survey = SurveySDK.getSurvey(serverId, surveyId)
            ?: throw Exception("Survey is not available")
        this.survey = survey
        this.selectedLanguage = survey.getDefaultLanguage()
    }

    constructor(survey: Survey) {
        this.survey = survey
        this.selectedLanguage = survey.getDefaultLanguage()
    }

    fun update(survey: Survey) {
        this.survey = survey
    }
}

File: examples_ios/MyApp/Sources/Controllers/ViewModels/Models/SurveyModel.swift
import ConfirmitMobileSDK
import Foundation

class SurveyModel {
  private(set) var survey: Survey
  var selectedLanguage: SurveyLanguage

  convenience init(serverId: String, surveyId: String) throws {
    guard let survey = try SurveySDK.getSurvey(serverId: serverId, surveyId: surveyId) else {
      throw AppError.general("Survey is not available")
    }

    try self.init(survey: survey)
  }

  init(survey: Survey) throws {
    self.survey = survey
    selectedLanguage = try survey.getDefaultLanguage()
  }

  func update(survey: Survey) {
    self.survey = survey
  }
}
